src/main/webapp/app/iris/course-chatbot/course-chatbot.component.ts

```diff
@@ -1,19 +1,23 @@
-import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { Component, OnChanges, SimpleChanges, input } from '@angular/core';
 import { ChatServiceMode, IrisChatService } from 'app/iris/iris-chat.service';
+import { IrisModule } from 'app/iris/iris.module';
+import { IrisBaseChatbotComponent } from 'app/iris/base-chatbot/iris-base-chatbot.component';
 
 @Component({
     selector: 'jhi-course-chatbot',
     templateUrl: './course-chatbot.component.html',
     styleUrl: './course-chatbot.component.scss',
+    standalone: true,
+    imports: [IrisModule, IrisBaseChatbotComponent],
 })
 export class CourseChatbotComponent implements OnChanges {
-    @Input() courseId?: number;
+    courseId = input<number>();
 
     constructor(public chatService: IrisChatService) {}
 
     ngOnChanges(changes: SimpleChanges): void {
         if (changes.courseId) {
-            this.chatService.switchTo(ChatServiceMode.COURSE, this.courseId);
+            this.chatService.switchTo(ChatServiceMode.COURSE, this.courseId());
```

Comment:
when using a signal as input you could get rid of ngOnChanges with an effect which runs everytime the input signal changes

```
constructor() {
    effect(() => this.chatService.switchTo(ChatServiceMode.COURSE, this.courseId()));
}
```